### PR TITLE
Spark 3.0: Reports bytesWritten and recordsWritten to metrics

### PR DESCRIPTION
**Pull request by izchen**
something like:

output_metrics1

output_metrics2

### CODE REVIEW

**Comment by izchen** [comment]
@rdblue @kbendick Could you help review? 😄

File: spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java [hunk 1]
```diff
@@ -492,9 +495,26 @@ public void doCommit(long epochId, WriterCommitMessage[] messages) {
...
     private final DataFile[] taskFiles;
+    private long bytesWritten = 0L;
+    private long recordsWritten = 0L;
 
     TaskCommit(DataFile[] taskFiles) {
       this.taskFiles = taskFiles;
+      for (DataFile dataFile : taskFiles) {
+        this.bytesWritten += dataFile.fileSizeInBytes();
+        this.recordsWritten += dataFile.recordCount();
+      }
```
Comment (rdblue) [inline spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:506]: Why store these as instance fields if this isn't going to expose them anywhere? I think these could be calculated in the report method.

File: spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java [hunk 2]
```diff
@@ -492,9 +495,26 @@ public void doCommit(long epochId, WriterCommitMessage[] messages) {
...
+    }
+
+    // Reports bytesWritten and recordsWritten to the Spark output metrics.
+    // Can only be called in executor.
+    void reportsToOutputMetrics() {
```
Comment (rdblue) [inline spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:511]: I think `reportOutputMetrics` is a slightly better name.

**Comment by izchen** [comment]
Thanks for your review, @rdblue !

File: spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java (hunk 1 above)
Comment (izchen) [reply spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java]: done

File: spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java (hunk 2 above)
Comment (izchen) [reply spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java]: done

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @izchen! I'll merge this when tests have passed. Do you want to also port this to Spark 3.1 and 3.2?

**Pull request merged**